### PR TITLE
Add "beatmap not downloaded / downloading" user states to enum

### PR DESCRIPTION
**Pull request by frenzibyte**
Dependency for server-side support.

### CODE REVIEW

File: osu.Game/Online/RealtimeMultiplayer/MultiplayerUserState.cs [hunk 1]
```diff
@@ -5,6 +5,16 @@ namespace osu.Game.Online.RealtimeMultiplayer
 {
     public enum MultiplayerUserState
     {
+        /// <summary>
+        /// The user has not downloaded the beatmap yet.
+        /// </summary>
+        BeatmapNotDownloaded,
```
Comment (peppy) [inline osu.Game/Online/RealtimeMultiplayer/MultiplayerUserState.cs:11]: have you looked how this is serialized, and whether inserting at the *start* of the enum will break everything?

Comment (frenzibyte) [reply osu.Game/Online/RealtimeMultiplayer/MultiplayerUserState.cs:11]: oh..., my bad.

**Comment by peppy** [comment]
Have you confirmed either way? I was asking because I was unsure - it may actually be okay depending on how the serialization is done.

**Comment by frenzibyte** [comment]
Leaving it the way I had has broken up everything due to different enum values between this version and the server's version (server's `Idle` becomes `BeatmapNotDownloaded` here, etc...), I think the states can be moved back to top when the server-side changes are merged.

**Comment by frenzibyte** [comment]
Closing as I'm going in another direction.

**Pull request closed without merging**